catalog/core/catalog-core-attributeregistry/src/main/java/ddf/catalog/data/impl/AttributeRegistryImpl.java

```diff
@@ -56,14 +61,50 @@ public boolean register(final AttributeDescriptor attributeDescriptor) {
     @Override
     public void deregister(final String name) {
         notNull(name, "The attribute name cannot be null.");
-
         attributeMap.remove(name);
     }
 
     @Override
     public Optional<AttributeDescriptor> lookup(final String name) {
         notNull(name, "The attribute name cannot be null.");
-
         return Optional.ofNullable(attributeMap.get(name));
     }
+
+    public void setBundleContext(BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
```

Comment:
Use `FrameworkUtil.getBundle().getBundleContext()` to safely acquire the bundleContext each time you want to use it.

https://osgi.org/javadoc/r4v43/core/org/osgi/framework/FrameworkUtil.html